api/osf_api.py

```diff
@@ -133,6 +134,27 @@ def delete_project(session, guid, user=None):
             n.get()
             n.delete()
 
+
+def create_custom_collection(session):
+    collections_url = '{}/v2/collections/'.format(session.api_base_url)
+
+    payload = {
+        'title': 'Selenium API Custom Collection',
+    }
+
+    session.post(collections_url, item_type='collections', attributes=payload)
+
+
+def delete_custom_collections(session):
+    collections_url = '{}/v2/collections/'.format(session.api_base_url)
+    data = session.get(collections_url)
+
+    for collection in data['data']:
+        if not collection['attributes']['bookmarks']:
+            collection_self_url = collections_url + collection['id']
+            requests.delete(collection_self_url, auth=session.auth)
```

Comment:
I suggest use `session.delete(...)` (if available).

`session` is a selenium object that contains the info (e.g. auth) about current user session while `requests` (https://pypi.org/project/requests/)  is a python library for making HTTP requests and receiving responses. Is there a specific reason for using either of the two?

Future TODO: we want to take a look the code base see how often we use the `session` and `requests`.  More importantly, is it just a random choice or one tries to solve another's problem (as a work-around).

Reply:
You are correct! `session.<action>` is used about 20 other times in this file. Meanwhile `requests` is only used 1 other time to catch an exception.